source/opt/decoration_manager.h

```diff
@@ -50,6 +50,8 @@ class DecorationManager {
   bool AreDecorationsTheSame(const ir::Instruction* inst1,
                              const ir::Instruction* inst2) const;
 
+  const ir::Instruction* GetDecoration(uint32_t id, uint32_t decoration) const;
```

Comment:
Need a method comment.

Reply:
A decoration enum *could* applied more than one time to a target.  In fact the LinkageAttributes is such a case.
You can both import *and* export a variable, yes?

So Suggest this should either return a vector, or better do a ForEachDecorationForIdOfKind(uint32_t id, uint32_t, std::function<void(const Instruction*)>)